components/server/src/iam/iam-session-app.ts

```diff
@@ -162,7 +162,7 @@ export class IamSessionApp {
                 primaryEmail: recent.primaryEmail,
                 lastSigninTime: new Date().toISOString(),
             });
-            await this.userService.updateUser(user.id, {
+            await this.userService.updateUser(SYSTEM_USER_ID, {
```

Comment:
How do we solve these things when we remove the subjectId as first argument but rely on the context instead?
Would this code run in the context of a system user?

Reply:
Yes, in this specific case: as this handler is run as SYSTEM_USER.

The current config ensures that both paths do match; once we are starting to remove the first arguments, some (most?) of these invocations should go away.